cEP-0022.md

```diff
@@ -0,0 +1,461 @@
+
+### Successive runs of `--green-mode`
+
+`lite-mode` should build upon the data from the previous runs to generate
+less error prone `.coafile` (less error prone over here indicates not likely
+to generate errors in the project), so that running `lite-mode` a number of
+times is still faster than Brute Force. We choose this method as it is highly
+unlikely that different files among the same file types have a different
+set of codestyle rules in a project.
+
+If a `.quickfile` is present in the directory, the successive
+runs will try to correct the configuration files (always acting upon
+`.coafile.new` over `.coafile` as a preference) assumming we intend to find
+a config file which is even more specific than the one already present. Absense
+of config files will build the config files from scratch.
+
+- Successive runs of `--lite-mode` will check for all settings and will
+  again
+  choose files at random. If it finds a conflicting setting with the config
+  file,
+  it will be appended to the `.quickfile`. If a setting is already present in
+  the
+  `.quickfile`, weights of the other values to bear settings will be appended.
+- Successive runs of **Brute Force** will assume that it is being run
+  on the results after a few `--lite-mode` runs and will only run
+  for the settings
+  provided in the `.quickfile` appending to the weights.
+
+### **The `--smart-mode`**
+
+It is clear that successive runs of the `--lite-mode` will overestimate
+the weights in the `.quickfile`, so it seems advantageous to run `--lite-mode`
+several times and then run the **Brute Force** for the settings dropped by
+`--lite-mode` which are now in `.quickfile` to generate correct value of the
+weight. This combination of `--lite-mode` and **Brute Force** is done
+collectively by `--smart-mode`.
+
+Every run will have the annoying question in the end whether the user wants
+to answer
+a few questions about the project to get better results unless changed upon
+by the provided tags. *(For more details check out the tags section)*
+
+# Learning from real world projects
+
+## Need to perform this task
+
+There can be endless number of possibilities and assumptions:
+
+- Files having similar kind of names can be having different codestyles.
+- Files in a given directory can be following different codestyles.
+- Different parts of files maybe even following a different codestyle depending
+  on function name which can also be handled by placing `ignore`s at
+  appropriate places in the file dict.
+
+  Such kind of functionality can only be
+  added to the bears themselves, but instead we stop our train of thought over
+  here.
+
+## Making Quickstart recognize common patterns
+
+We save ourselves the additional work for building some feature that may
+hardly ever be used by any org, instead we stop assumming and start
+implementing and learning from our available resources.
+
+We now choose a list of orgs for which we try to test our `green-mode`
```

Comment:
It's nice to get down to select these orgs now, because we have to choose some repos as our benchmarks anyways.